### PR TITLE
Fix issue DPTP-3553 Add a makefile target to generate HyperShift deployment

### PR DESCRIPTION
**Pull request by bear-redhat**
Add a makefile target to generate HyperShift deployment

Multi-versions are not supported. We always use `latest`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -412,3 +412,35 @@ secret-config-updater:
...
+	chmod +x ${TMPDIR}/hypershift-cli/hypershift
+
+	${TMPDIR}/hypershift-cli/hypershift \
+		install \
+		--oidc-storage-provider-s3-bucket-name=hypershift-oidc-provider \
+		--oidc-storage-provider-s3-credentials=${MGMT_AWS_CONFIG_PATH} \
+		--oidc-storage-provider-s3-region=us-east-1 \
+		--hypershift-image=registry.ci.openshift.org/ci/hypershift-cli:latest \
+		--enable-uwm-telemetry-remote-write=false \
+		render > $(CURDIR)/clusters/hive/hypershift/hypershift-install.yaml
```
Comment (hongkailiu) [inline Makefile:445]:
```suggestion
		render > clusters/hive/hypershift/hypershift-install.yaml
```

File: Makefile [hunk 2]
```diff
@@ -412,3 +412,35 @@ secret-config-updater:
...
+__check_defined = \
+	$(if $(value $1),, \
+		$(error Undefined environment variable $1$(if $2, ($2))))
+
+generate-hypershift-deplyment:
+	@:$(call check_defined, MGMT_AWS_CONFIG_PATH)
+	@:$(call check_defined, TMPDIR)
+
+	mkdir -p ${TMPDIR}/hypershift-cli
+	oc image extract registry.ci.openshift.org/ci/hypershift-cli:latest --path /usr/bin/hypershift:${TMPDIR}/hypershift-cli --confirm
```
Comment (hongkailiu) [inline Makefile:435]: Isn't the source image in quay.io?

File: Makefile [hunk 3]
```diff
@@ -412,3 +412,35 @@ secret-config-updater:
...
+	chmod +x ${TMPDIR}/hypershift-cli/hypershift
+
+	${TMPDIR}/hypershift-cli/hypershift \
+		install \
+		--oidc-storage-provider-s3-bucket-name=hypershift-oidc-provider \
+		--oidc-storage-provider-s3-credentials=${MGMT_AWS_CONFIG_PATH} \
```
Comment (hongkailiu) [inline Makefile:441]: Will the generated yaml contain the credentials as plain text?

**Review by hongkailiu — APPROVED** [review]

File: Makefile [hunk 4]
```diff
@@ -412,3 +412,35 @@ secret-config-updater:
...
+check_defined = \
+	$(strip $(foreach 1,$1, \
+		$(call __check_defined,$1,$(strip $(value 2)))))
+__check_defined = \
+	$(if $(value $1),, \
+		$(error Undefined environment variable $1$(if $2, ($2))))
+
+generate-hypershift-deplyment:
+	@:$(call check_defined, MGMT_AWS_CONFIG_PATH)
+	@:$(call check_defined, TMPDIR)
```
[the remark this answers is not in the document]
Comment (bear-redhat) [reply Makefile:432]: Actually I dont think we shall even define the default value for `TMPDIR` here... This is a OS-defined variable. If it is absent, then the program should abort.

File: Makefile (hunk 3 above)
Comment (bear-redhat) [reply Makefile:441]: Thats a valid point. I will check.

**Pull request merged**